demo/angular-ngFor.ts

```diff
@@ -1,66 +1,122 @@
 /**
  * Example using Angular ngFor to loop through items and create DOM items
  */
- import { Component, AfterViewInit, OnChanges, SimpleChanges, Input, ChangeDetectionStrategy } from '@angular/core';
+import {
+  Component,
+  AfterViewInit,
+  Input,
+  ViewChildren,
+  QueryList,
+} from "@angular/core";
 
- import { GridStack, GridStackWidget } from 'gridstack';
- import 'gridstack/dist/h5/gridstack-dd-native';
- 
- @Component({
-   selector: 'app-angular-ng-for-test',
-   template: `
-     <button (click)="add()">add item</button><button (click)="delete()">remove item</button><button (click)="change()">modify item</button>
-     <div class="grid-stack">
-       <!-- using angular templating to create DOM, otherwise an easier way is to simply call grid.load(items) -->
-       <div *ngFor="let n of items; let i = index" class="grid-stack-item" [attr.gs-w]="n.w" [attr.gs-h]="n.h" [attr.gs-x]="n.x" [attr.gs-y]="n.y">
-         <div class="grid-stack-item-content">item {{i}}</div>
-       </div>
-     </div>
-     `,
-   // gridstack.min.css and other custom styles should be included in global styles.scss
- 
-   // tell Angular only @Input will change - doesn't help
-   // changeDetection: ChangeDetectionStrategy.OnPush
- })
- export class AngularNgForTestComponent implements AfterViewInit, OnChanges {
+import { GridStack, GridStackWidget } from "gridstack";
+import "gridstack/dist/h5/gridstack-dd-native";
+import { Subject, zip } from "rxjs";
+
+@Component({
+  selector: "app-angular-ng-for-test",
+  template: `
+    <button (click)="add()">add item</button
+    ><button (click)="delete()">remove item</button
+    ><button (click)="change()">modify item</button>
+    <div class="grid-stack">
+      <!-- using angular templating to create DOM, otherwise an easier way is to simply call grid.load(items) -->
+      <div
+        *ngFor="let n of items; let i = index; trackBy: identify"
+        [id]="i"
+        class="grid-stack-item"
+        [attr.gs-w]="n.w"
+        [attr.gs-h]="n.h"
+        [attr.gs-x]="n.x"
+        [attr.gs-y]="n.y"
+        #gridStackItem
+      >
+        <div class="grid-stack-item-content">item {{ i }}</div>
+      </div>
+    </div>
+  `,
+  styles: [
+    `
+      // !!!IMPORTANT!!! Import this through your styles.scss or angular.json! This is just for demo purposes
+      :host {
+        ::ng-deep {
+          @import "gridstack/dist/gridstack.min";
+        }
+      }
+      .grid-stack {
+        margin: 15px;
+        background-color: #d66666;
+        min-height: calc(100vh - 50px);
+      }
+
+      .grid-stack-item {
+        border: 1px solid #d1d1d1;
+      }
+    `,
+  ],
+})
+export class AngularNgForTestComponent implements AfterViewInit {
+  @ViewChildren("gridStackItem") gridstackItems: QueryList<any>;

```

Comment:
Yes, probably. OnChanges is only triggered, when something comes in from outside. So this component is used and e.G. @Input is changed. But it has to be a new reference of the array for ngOnChanges to be triggered. But even for that you might want to handle the change with a setter of the @Input because then you know exactly that only that Input changed.

Reply:
I know the setter trick, and that the array itself has to change (not just the content) which is what I did... bur the onChange was still not called - maybe because it is it's method and angular prevent change detection during that time.

Reply:
No, in the previous version you just did `this.items = [...this.items, {x: 1, y: 6, w: 3}];` which won´t trigger ngOnChanges. ngOnChanges is only triggered if you pass in from outside like this:

<app-angular-ng-for-test [items]="[{x: 1, y: 6, w: 3}]"></app-angular-ng-for-test>